packages/angular/build/src/tools/vite/plugins/setup-middlewares-plugin.ts

```diff
@@ -55,6 +55,57 @@ interface AngularSetupMiddlewaresPluginOptions {
   ssrMode: ServerSsrMode;
   resetComponentUpdates: () => void;
   projectRoot: string;
+  allowedHosts: true | string[];
+  devHost: string;
+}
+
+function extractHostname(hostHeader: string | undefined): string {
+  if (!hostHeader) {
+    return '';
+  }
+
+  // Remove port if present (e.g., example.com:4200)
+  const idx = hostHeader.lastIndexOf(':');
+  if (idx > -1 && hostHeader.indexOf(']') === -1) {
+    // Skip IPv6 addresses that include ':' within brackets
+    return hostHeader.slice(0, idx).toLowerCase();
+  }
+
+  return hostHeader.toLowerCase();
+}
+
+function html403(hostname: string): string {
```

Comment:
Optional NIT: Would it be possible to move this in a new file? (See other middlewares)